core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -207,7 +210,23 @@ private[spark] class Executor(
 
         // Run the actual task and measure its runtime.
         taskStart = System.currentTimeMillis()
-        val value = task.run(taskAttemptId = taskId, attemptNumber = attemptNumber)
+        var succeeded: Boolean = false
+        val value = try {
+          val value = task.run(taskAttemptId = taskId, attemptNumber = attemptNumber)
+          succeeded = true
+          value
+        } finally {
+          // Release managed memory used by this task
+          val freedMemory = taskMemoryManager.cleanUpAllAllocatedMemory()
+          if (succeeded && freedMemory > 0) {
```

Comment:
maybe we want to enforce so that even in the case of failures, memory should be cleaned up?

also would be great to not have two different code path in executor vs dagscheduler